compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -2648,6 +2648,100 @@ declare_lint! {
     };
 }
 
+declare_lint! {
+    /// The `fuzzy_provenance_casts` lint detects an `as` cast between an integer
+    /// and a pointer.
+    ///
+    /// ### Example
+    ///
+    /// ```rust
+    /// #![feature(strict_provenance)]
+    /// #![warn(fuzzy_provenance_casts)]
+    ///
+    /// fn main() {
+    ///     let _dangling = 16_usize as *const u8;
+    /// }
+    /// ```
+    ///
+    /// {{produces}}
+    ///
+    /// ### Explanation
+    ///
+    /// This lint is part of the strict provenance effort, see [issue #95228].
+    /// Casting an integer to a pointer is considered bad style, as a pointer
+    /// contains, besides the *address* also a *provenance*, indicating what
+    /// memory the pointer is allowed to read/write. Casting an integer, which
+    /// doesn't have provenance, to a pointer requires the compiler to assign
+    /// (guess) provenance. The compiler assigns "all exposed valid" (see the
+    /// docs of [`ptr::from_exposed_addr`] for more information about this
+    /// "exposing"). This penalizes the optimiser and is not well suited for
+    /// dynamic analysis/dynamic program verification (e.g. Miri or CHERI
+    /// platforms).
+    ///
+    /// It is much better to use [`ptr::with_addr`] instead to specify the
+    /// provenance you want. If using this function is not possible because the
+    /// code relies on exposed provenance then there is (as an escape hatch)
+    /// [`ptr::from_exposed_addr`].
+    ///
+    /// This might (or might not) become a [future-incompatible] lint or hard
+    /// error in a future rust edition. See [issue #95228] for more details.
```

Comment:
"rust" -> "Rust"?

Reply:
Actually I think it is very unlikely that it is going to become a future-incompatible lint. What I meant is that it could become an edition lint in the future, but since the whole strict-pointer experiment is very young and still unstable, even that is quite uncertain. So I think it is better to remove that whole section for now.